File: 15_09_2022/1370._Increasing_Decreasing_String.cpp
class Solution {
public:
    string sortString(string s) {
        map<char,int> d;
        string ans="";
        for(auto &x:s)
        {
            d[x]++;
        }
        
        int l=s.length();
        while(l>0)
        {
            for(auto &x:d)
            {
                if(x.second>=1)
                {
                    ans+=x.first;
                    x.second--;
                    l--;
                }
            }
            
            for(auto k='z';k>='a';k--)
            {
                if(d[k]>=1)
                {
                    ans+=k;
                    d[k]--;
                    l--;
                }
            }
        }
        return ans;
    }
};

File: 15_09_2022/1370._Increasing_Decreasing_String.py
class Solution:
    def sortString(self, s: str) -> str:
        d=[0]*26
        ans=""
        for i in s:
            i=ord(i)-97
            d[i]+=1
        print(d)
        l=len(s)
        print(l)
        while(l>0):
            for i in range(0,26):
                if(d[i]>=1):
                    ans+=chr(i+97)
                    d[i]-=1
                    l-=1

            for i in range(25,-1,-1):
                if(d[i]>=1):
                    ans+=chr(i+97)
                    d[i]-=1
                    l-=1
        return ans
                  
            
                    
            
                
        